### PR TITLE
implements shield utils and CFR interactions (re #271)

### PR DESCRIPTION
**Pull request by raymak**

### CODE REVIEW

**Comment by raymak** [comment]
one issue left here is that the survey cannot get the userid (and condition variables) as input

File: phase2/addon/lib/experiment.js [hunk 1]
```diff
@@ -257,6 +257,16 @@ const stages = {
 };
 
 function end(){
+
```
Comment (gregglind) [inline phase2/addon/lib/experiment.js]: backwards-compatible handling

File: phase2/addon/lib/experiment.js [hunk 2]
```diff
@@ -286,7 +287,8 @@ function end(){
     });
 
     // delay to give some time for the remaining message queue to be flushed
-    setTimeout(function() {require("./utils").selfDestruct("end");}, prefs["experiment.modes.end.delay"]);
+    if (!prefs["death_handled_by_xutils"] && !prefs["experiment.live_forever"])
+      setTimeout(function() {require("./utils").selfDestruct("end");}, prefs["experiment.modes.end.delay"]);
```
Comment (gregglind) [inline phase2/addon/lib/experiment.js]: backwards compatible for death

File: phase2/addon/lib/index.js [hunk 3]
```diff
@@ -120,7 +120,7 @@ exports.onUnload = function(reason){
 
   require('./sender').flush();
 
-  if (reason == "uninstall" && prefs["cleanup_on_death"])
+  if (!prefs["cleanup_handled_by_xutils"] && reason == "uninstall" && prefs["cleanup_on_death"])
```
Comment (gregglind) [inline phase2/addon/lib/index.js]: backwards compat

File: phase2/addon/lib/shield-studies-facade.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ let variationsMod = {
     'all': function () {}
   },
   isEligible: function () {return true},
-  cleanup: function () {}
+  cleanup: function () {require('./utils').cleanUp({reset: true})}
```
Comment (gregglind) [inline phase2/addon/lib/shield-studies-facade.js]: I think this is right :)

+1 r+

**Pull request merged**